modules/core/src/main/java/org/apache/ignite/internal/client/impl/connection/GridClientConnectionManagerAdapter.java

```diff
@@ -518,7 +518,10 @@ protected GridClientConnection connect(@Nullable UUID nodeId, InetSocketAddress
 
         closeIdle();
 
-        conn.close(FAILED, false);
+        if (conn instanceof GridClientNioTcpConnection)
```

Comment:
whats a purpose of such code ? lets change org.apache.ignite.internal.client.impl.connection.GridClientConnection#close(GridClientConnectionCloseReason reason, boolean waitCompletion);
into 
close(GridClientConnectionCloseReason reason, boolean waitCompletion, @Nullable Throwable cause)
and no additional cast would be needed here.

Reply:
Fixed.